spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/quota/RedisQuotaFilter.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.filter.quota;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.validation.constraints.Min;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jetbrains.annotations.NotNull;
```

Comment:
I think it is a mistake import.

Reply:
Its the same that the RedisRateLimiter use, but yes think it would be better to switch to slf4j

Reply:
I talk about "org.jetbrains.annotations" :)

Reply:
Indeed there is the same in RedisRateLimiter I think it was also a mistake in RedisRateLimiter and "javax.validation.constraints.NotNull" should be used.

Reply:
i'll change that